libparsec/crates/client/tests/unit/certif/share_realm.rs

```diff
@@ -0,0 +1,328 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use libparsec_client_connection::{test_register_send_hook, test_register_sequence_of_send_hooks};
+use libparsec_protocol::authenticated_cmds;
+use libparsec_tests_fixtures::prelude::*;
+use libparsec_types::prelude::*;
+
+use crate::certif::{CertifShareRealmError, CertificateBasedActionOutcome};
+
+use super::utils::certificates_ops_factory;
+
+#[parsec_test(testbed = "minimal")]
+async fn ok(env: &TestbedEnv) {
+    let (env, realm_id) = env.customize_with_map(|builder| {
+        builder.new_user("bob");
+        builder
+            .new_realm("alice")
+            .then_do_initial_key_rotation()
+            .map(|event| event.realm)
+    });
+    let alice = env.local_device("alice@dev1");
+    let ops = certificates_ops_factory(&env, &alice).await;
+
+    ops.add_certificates_batch(
+        &env.get_common_certificates_signed(),
+        &[],
+        &[],
+        &env.get_realms_certificates_signed(),
+    )
+    .await
+    .unwrap();
+
+    let keys_bundle = env.get_last_realm_keys_bundle(realm_id);
+    let keys_bundle_access = env.get_last_realm_keys_bundle_access_for(realm_id, &alice.user_id());
+    test_register_sequence_of_send_hooks!(
+        &env.discriminant_dir,
+        {
+            move |req: authenticated_cmds::latest::realm_get_keys_bundle::Req| {
+                p_assert_eq!(req.key_index, Some(1));
+                p_assert_eq!(req.realm_id, realm_id);
+
+                authenticated_cmds::latest::realm_get_keys_bundle::Rep::Ok {
+                    key_index: 1,
+                    keys_bundle,
+                    keys_bundle_access,
+                }
+            }
+        },
+        {
+            move |req: authenticated_cmds::latest::realm_share::Req| {
+                p_assert_eq!(req.key_index, 1);
+                authenticated_cmds::latest::realm_share::Rep::Ok
+            }
+        },
+    );
+
+    let res = ops
+        .share_realm(
+            realm_id,
+            "bob".parse().unwrap(),
+            Some(RealmRole::Contributor),
+        )
+        .await
+        .unwrap();
+
+    p_assert_matches!(res, CertificateBasedActionOutcome::Uploaded { .. });
```

Comment:
Can you verify that `bob` have access to the realm ?

Reply:
I don't think this is the job of this test, because I should create `2` ops, `1` for `alice` and `1` for `bob`, and it is also tested by high level tests:
For example the `CLI` test